sdk/eventhub/Azure.Messaging.EventHubs/tests/Processor/EventProcessorLiveTests.cs

```diff
@@ -0,0 +1,907 @@
+                    // in the order they were sent.
+
+                    foreach (var partitionId in partitionIds)
+                    {
+                        Assert.That(allReceivedEvents.TryGetValue(partitionId, out var partitionReceivedEvents), Is.True, $"{ partitionId }: there should have been a set of events received.");
+                        Assert.That(partitionReceivedEvents.Count, Is.EqualTo(expectedEvents[partitionId].Count), $"{ partitionId }: amount of received events should match.");
+
+                        var index = 0;
+
+                        foreach (var receivedEvent in partitionReceivedEvents)
+                        {
+                            Assert.That(receivedEvent.IsEquivalentTo(expectedEvents[partitionId][index]), Is.True, $"{ partitionId }: the received event at index { index } did not match the sent set of events.");
+                            ++index;
+                        }
+                    }
+
+                    Assert.That(allReceivedEvents.Keys.Count, Is.EqualTo(partitionIds.Count()));
+                }
+            }
+        }
+
+        /// <summary>
+        ///   Verifies that the <see cref="EventProcessor" /> is able to
+        ///   connect to the Event Hubs service and perform operations.
+        /// </summary>
+        ///
+        [Test]
+        public async Task PartitionProcessorProcessEventsAsyncIsCalledWithNoEvents()
+        {
+            await using (var scope = await EventHubScope.CreateAsync(1))
+            {
+                var connectionString = TestEnvironment.BuildConnectionStringForEventHub(scope.EventHubName);
+
+                await using (var client = new EventHubClient(connectionString))
+                {
+                    var receivedEventSets = new ConcurrentBag<IEnumerable<EventData>>();
+
+                    // Create the event processor hub to manage our event processors.
+
+                    var hub = new EventProcessorHub
+                        (
+                            EventHubConsumer.DefaultConsumerGroupName,
+                            client,
+                            onProcessEvents: (partitionContext, checkpointManager, events, cancellationToken) =>
+                                receivedEventSets.Add(events)
+                        );
+
+                    hub.AddEventProcessors(1);
+
+                    // Start the event processors.
+
+                    await hub.StartAllAsync();
+
+                    // Make sure the event processors have enough time to stabilize.
+                    // TODO: we'll probably need to extend this delay once load balancing is implemented.
+
+                    await Task.Delay(5000);
+
+                    // Stop the event processors.
+
+                    await hub.StopAllAsync();
+
+                    // Validate results.
+
+                    Assert.That(receivedEventSets, Is.Not.Empty);
+
+                    foreach (var eventSet in receivedEventSets)
```

Comment:
Since you're not looking at any particular identifier for which is empty, maybe consider refactoring to something like:

```csharp
Assert.That(receivedEventSets.Any(set => (set == null || !set.Any())), Is.False);
```

Reply:
Good suggestion. Haven't used `Any` before.